### PR TITLE
*: handle MULTIPOINT containing empty elements

### PR DESCRIPTION
**Pull request by otan**
To handle empty, we convert MultiPoint to inherit geom2 and use the ends
array to determine whether there are empty "holes" in the flat coords
array. This involves a bunch of overriding geom2 methods with geom1-like
methods.

Updated the WKB/EWKB methods to reflect the changes.

### CODE REVIEW

File: point.go [hunk 1]
```diff
@@ -24,6 +26,22 @@ func NewPointFlat(l Layout, flatCoords []float64) *Point {
 	return g
 }
 
+// NewPointFlatMaybeEmpty returns a new point, checking whether the point may be empty
+// by checking wther all the points are NaN.
+func NewPointFlatMaybeEmpty(layout Layout, flatCoords []float64) *Point {
+	isEmpty := true
+	for _, coord := range flatCoords {
+		if !math.IsNaN(coord) {
```
Comment (twpayne) [inline point.go:34]: Here the code use `math.IsNaN`, but really to represent an empty point we're using a specific NaN value for all elements of `flatCoords`, so this test is a little too eager to set `isEmpty` to `true`. I would replace it with a test that checks that all elements of `flatCoords` are the specific NaN that we use.

Comment (otan) [reply point.go:34]: we can't test that as NaN != NaN :(

Comment (otan) [reply point.go:34]: (well, without doing some complex hex stuff: https://stackoverflow.com/questions/21869571/nan-different-from-nan-in-go)

Comment (twpayne) [reply point.go:34]: It is possible: https://play.golang.org/p/w7Pov-SPuSJ

File: multipoint.go [hunk 2]
```diff
@@ -40,14 +73,55 @@ func (g *MultiPoint) MustSetCoords(coords []Coord) *MultiPoint {
...
+}
+
 // SetCoords sets the coordinates.
 func (g *MultiPoint) SetCoords(coords []Coord) (*MultiPoint, error) {
-	if err := g.setCoords(coords); err != nil {
-		return nil, err
+	g.flatCoords = nil
+	g.ends = nil
+	for _, c := range coords {
+		if len(c) > 0 {
```
Comment (twpayne) [inline multipoint.go:93]: Here we implicitly have two representations for an empty point: either `nil` or `[]float64{}` (a `Coord` with no elements). I think we should only use `nil` and `[]float64{}` should return a `ErrLayoutMismatch` error instead.

Comment (otan) [reply multipoint.go]: i buy that argument! done.

**Pull request merged**